terminal/src/identity.rs

```diff
@@ -0,0 +1,63 @@
+use librad::profile::Profile;
+
+use radicle_common::fmt;
+use radicle_common::profile;
+
+use crate as term;
+
+/// Identity formatter that takes a profile and displays it `peer_id` and
+/// `username` depending on the configuration.
+///
+/// If short display is requested, `peer_id` is printed in its compact format
+/// e.g. `hynddpk…uf4qwge`. If `styled` is requested, `peer_id` and `username`
+/// are printed using the terminal's styled formatters.
+pub struct Formatter<'a> {
+    profile: &'a Profile,
+    shorten: bool,
+    style: bool,
+}
+
+impl<'a> Formatter<'a> {
+    pub fn new(profile: &'a Profile) -> Self {
+        Self {
+            profile,
+            shorten: false,
+            style: false,
+        }
+    }
+
+    pub fn shorten(mut self) -> Self {
+        self.shorten = true;
+        self
+    }
+
+    pub fn style(mut self) -> Self {
+        self.style = true;
+        self
+    }
```

Comment:
For methods that move and return `self`, I tend to prefer using adjectives, to communicate the fact that it isn't 'doing' (mutating) anything to `self`. So for example the above would be `short` and `styled`.

Reply:
Hehe, had that before and switched to verbs. But yeah, agreed!

Reply:
Haha cool. Yeah another example is when doing verification: `Foo::verify(&self) -> bool` but `Foo::verified(self) -> VerifiedFoo`